### PR TITLE
Port fix for enumerating directories on XFS

### PR DESCRIPTION
**Pull request by danmosemsft**
Porting fix for https://github.com/dotnet/corefx/issues/29095

cc @JeremyKuhne

### CODE REVIEW

**Comment by danmosemsft** [comment]
cc @weshaggard 

One workaround to bootstrap our way out of this is to build on some other filesystem.

File: src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs [hunk 1]
```diff
@@ -44,10 +44,14 @@ namespace System.IO.Enumeration
                 // We know it's a directory.
                 isDirectory = true;
             }
-            else if ((directoryEntry.InodeType == Interop.Sys.NodeType.DT_LNK)
-                && Interop.Sys.Stat(entry.FullPath, out Interop.Sys.FileStatus targetStatus) >= 0)
+            else if ((directoryEntry.InodeType == Interop.Sys.NodeType.DT_LNK
+                || directoryEntry.InodeType == Interop.Sys.NodeType.DT_UNKNOWN)
+                && (Interop.Sys.Stat(entry.FullPath, out Interop.Sys.FileStatus targetStatus) >= 0
+                    || Interop.Sys.LStat(entry.FullPath, out targetStatus) >= 0))
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs]: As discussed in the original thread, this LStat appears to be unnecessary and will just be a nop but overhead in the case where Stat fails, but it was there in 2.0 in this use case and shouldn't hurt anything, so we can look at removing it post-2.1.

**Comment by JeremyKuhne** [comment]
Master PR is #29502

**Comment by danmosemsft** [comment]
OSX failure is https://github.com/dotnet/corefx/issues/21037

**Pull request merged**